internal/core/encoding.go

```diff
@@ -191,6 +193,26 @@ func convertToInt(propertyName string, untypedValue any) (int64, error) {
 	}
 }
 
+func convertToJSON(propertyName string, untypedValue any) (any, error) {
+	untypedMap, ok := untypedValue.(map[any]any)
+	if !ok {
+		return untypedValue, nil
```

Comment:
todo: If this line is hit it is a programming error, please either return an error, or let it panic.  Or explicitly test for nil if that is what is really wanted here.

Reply:
This is required because when maps are decoded from cbor they are of type `map[any]any` and must be converted to `map[string]any`. Any other value is also valid JSON so we return early if they are not maps.

Reply:
Ah I see what you mean, can you please document this with a comment?  It looks like defensive coding otherwise

Reply:
done